ooni/deck/store.py

```diff
@@ -51,30 +73,54 @@ def update_url_lists(self, country_code):
             desc_fname = "citizenlab-test-lists_{0}.desc".format(cc)
 
             out_file = self.path.child("data").child(data_fname)
-            out_fh = out_file.open('w')
-            with in_file.open('r') as in_fh:
-                csvreader = csv.reader(in_fh)
-                csvreader.next()
-                for row in csvreader:
-                    out_fh.write("%s\n" % row[0])
-            out_fh.close()
+            write_txt_from_csv(in_file, out_file,
+                lambda row: "{}\n".format(row[0])
+            )
 
             desc_file = self.path.child("descriptors").child(desc_fname)
-            with desc_file.open('w') as out_fh:
-                if cc == "global":
-                    name = "List of globally accessed websites"
-                else:
-                    # XXX resolve this to a human readable country name
-                    country_name = cc
-                    name = "List of websites for {0}".format(country_name)
-                json.dump({
-                    "name": name,
-                    "filepath": out_file.path,
-                    "last_updated": timestampNowISO8601UTC(),
-                    "id": "citizenlab_{0}_urls".format(cc),
-                    "type": "file/url"
-                }, out_fh)
+            if cc == "global":
+                name = "List of globally accessed websites"
+            else:
+                # XXX resolve this to a human readable country name
+                country_name = cc
+                name = "List of websites for {0}".format(country_name)
+            write_descriptor(desc_file, name,
+                             "citizenlab_{0}_urls".format(cc),
+                             out_file.path,
+                             "file/url")
+
         self._cache_stale = True
+        yield defer.succeed(None)
+
+    @defer.inlineCallbacks
+    def update_tor_bridge_lines(self, country_code):
+        in_file = self.resources.child("tor-bridges").child(
+            "tor-bridges-ip-port.csv"
+        )
+        if not in_file.exists():
+            yield check_for_update(country_code)
+
+        data_fname = "tor-bridge-lines.txt"
+        desc_fname = "tor-bridge-lines.desc"
+        out_file = self.path.child("data").child(data_fname)
+
+        def format_row(row):
+            host, port, nickname, protocol = row
+            if protocol in ("ssh", "https", "port-444"):
+                return "{}:{}\n".format(host, port)
+            return "{} {}:{}\n".format(protocol, host, port)
```

Comment:
The reason for dropping the protocol at the beginning is that we care to indicate the protocol only in the case that it's a pluggable transport type.

In other cases we should just treat it as a non pluggable transport bridge and go ahead with a regular bridge test.